### ISSUE TITLE
Talent description inconsistencies

### ISSUE THREAD

**Issue by ak2yny** [ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance#50]
The talent, passive, powerup value descriptions currently don't follow a strict rule concerning the case. Some words are upper case for no apparent reason. We decided to make these descriptions follow a normal sentence rule, which is what lists also follow: https://www.gsbe.co.uk/grammar-lists.html
We have variables in descriptions, such as `$DMG`, which are always upper case ("Damage"). They are also used in the descriptions, and I haven't seen them at the beginning of a sentence, yet. They could be seen as names, which are upper case, or they could be changed to lower case.
A few variables are abbreviated, but there are also non-variable words which are abbreviated, such as `sec` and `max`. They should all be changed to full words ("seconds", "maximal", where we may have to change "seconds" to "second(s)").

Files to change (incomplete):
- [ ] all [talent](https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/tree/master/data/talents) files
- [ ] [shared_talents](https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/blob/master/data/shared_talents.engb.xml)
- [ ] [shared_powerups](https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/blob/master/data/shared_powerups.xmlb)
- [ ] [items](https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/blob/master/data/items.engb.xml)
- [ ] [team_bonus](https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/blob/master/data/team_bonus.engb.xml)

And possibly variables:
- [ ] [strings](https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/blob/master/data/strings.engb.xml)

-----------------------------------------

While doing talents, it may be a good opportunity to check for #10 .
Some costume passives are identical but have a different title, or description. They should probably be made the same, if the bonus is really identical. We may have to decide if we want to make it a long or short description. Long: `+10% damage to all powers`. Short: `+10% to powers`.
Some of the passives have already been fixed by @JordanLeich .

**Comment by techurious** [comment]
Need to be careful about this because some talent descriptions can become too long and look weird in the talent details screen.

**Comment by techurious** [comment]
Regarding the identical passives with different names: Need to see if the different name is unique to the character (only one instance) or if it's mixed (some characters use one, others use another for no obvious reason). If it's mixed, then it's probably worth it to rename them all to one standard. If only one character has unique name for its passive, perhaps we should leave it as it is, because I imagine Raven wanted to be creative with names of the passives for some characters. Undoing that creativity goes against Raven's vision and we shouldn't really mess with that.

**Comment by ak2yny** [comment]
So, for the inconsistent descriptions, I went for the wordy option. It should become clear why, when you look at following example from Iron Man:
```
Power Damage
Increases power damage.
```
And then from Thing:
```
Power Damage
Increases the damage of all powers.
```
And to make it completely clear, my explanation: Iron Man's description only has one more word,  and it doesn't really tell you anything more than the title (Power Damage). When you take the Thing's description, you don't have some questions, like "which power?" left.
People rarely look at the description, but if they do, they should get a sentence with more details.

**Comment by ak2yny** [comment]
A few notes about changes that were done along the lines of inconsistencies:
- Moon Knight's Crescent weapon descriptions are supposed to be upper case, since they are signature weapons which are written like names: [Wiki article](https://marvel.fandom.com/wiki/Marc_Spector_(Earth-616))
(There is also an instance where darts was changed to carts)
Update: 8b7c54710db6000a1734b6969a6c41bf05b3d7aa, be05b89cabc57ba1e44ef6bf1d3da8e529f1220c
- The `$DMG` variable was changed to lowercase in strings.engb. Changing that to lowercase text might even decrease consistency, and I left the variable intentionally.
Update: 8b7c54710db6000a1734b6969a6c41bf05b3d7aa, be05b89cabc57ba1e44ef6bf1d3da8e529f1220c
- The value descriptions usually have a modifier to convert the variable values into percent or something (`:f`, `:s`, `:p`, `:x`). Documented [here](https://docs.google.com/spreadsheets/d/1MusLD42l0cxwuNjyKAL9P7W1KlDub6BXqO-EoYfP04I/edit#gid=1679310935). We should generally leave them alone, except if we know that it is wrong. Moon Knight's Energy Regeneration passive was changed to `+` + `:f` + `%`, which will display value `1.05` as `+1.05%`, I believe (not 100% sure with the fraction). The nature of the passive code is that 5 is the actual increase while 1 (100) is the normal rate. So, 1.05 means a 105% regen rate, which is a 5% increase. The description should be `+5%`. The original code, `:s`, displays this correctly as it stands for scale (1.05 scales 100% to 105%), and adds `+` and `%` (instead of * 1.05 it displays `+5%` which is relative to 100% and means `105%` which is equal to * 1.05).
Update: be05b89cabc57ba1e44ef6bf1d3da8e529f1220c
- I think that less repetition is an enrichment to grammar, as long as the sentences are correct and clear. This at least in my native language, not sure about English. So if the title says "Experience Points", it would be better to read, "The amount of experience required to next level", than "The amount of experience points required to next level", I think. I don't see an inconsistency if it doesn't say "points". The original title was XP, which might mean e**xp**erience or **e**xperience **p**oints...
Update: 8b7c54710db6000a1734b6969a6c41bf05b3d7aa

-------------------------------------

I wasn't sure about some damage descriptions. There is a title, a description, and a value description. I think we should leave the value description as short as possible. Since title and/or description already include the specifics of the damage, we don't necessarily have to add them to the value description, do we? Adding general types, like energy damage, gun damage, weapon damage, melee damage, etc. makes more sense and is something that is better to understand than eg. Mjolnir damage. I think where the devs left the type/specifics out of value descriptions, we should leave it that way.

Example:
Title (descname): "Crescent Damage"
Description: "Increases the amount of $DMG done with Crescent powers."
Value description: "+5% Crescent damage" (The devs just wrote "+5% damage".)

**Comment by techurious** [comment]
> Example:
Title (descname): "Crescent Damage"
Description: "Increases the amount of $DMG done with Crescent powers."
Value description: "+5% Crescent damage" (The devs just wrote "+5% damage".)

Crescent damage is not a damage type. Physical, Energy, Elemental, Fire, Cold, etc are all damage types (damagetype in powerstyle) that need to be specified.

 Crescent Damage used in this case refers to attacks that use powerup_tag similar to Cap's Shield attacks or Cyclops' Laser Beam attacks. They have nothing to do with damagetype. Since the passive merely boosts the damage of the existing powers instead of adding its own damage on top of it as a separate attack, I think it's perfectly fine to refer to it as Damage because it references to damage of tagged powers (that have their damage type specified usually). Can't really say Power Damage either because not all powers are crescent powers. The most accurate way to reference it would be Crescent Power Damage but it's needlessly wordy. The title and description tell you what you need to know, and the values simply boost raw damage with no additional filters.

Case in point, Cyclops. He has a passive that boosts his Laser Beam damage. Likewise it references to fightmoves that use the specific powerup_tag. His beam powers can either deal energy damage or fire damage and probably physical damage in some cases so can't really mention a single damage type. And saying Beam Damage could be confusing because there are specifically powers that use attacktype = beam but his Optic Blast for example uses a simple punch trigger instead.

**Comment by ak2yny** [comment]
Thanks, that's exactly what I meant.
The reason we need anything after the values (`+5%`) is that usually there are other values and we need to specify which value means what.

Remember that most players know nothing about attack types. For them, most of Cyclop's attacks are beams. The devs put it there, so my opinion is to leave it. Personally, I don't have inconsistency issues when Cyclops has it and Moon Knight doesn't. Either way is fine with me (all such damages with additional word, all only damage, or leave it as it was).
Just mention one more time: Personally, `beam damage` sounds good to me, `Crescent damage` doesn't.

**Comment by ak2yny** [comment]
Made more updates: 2142e54498dd5a4ea8025b5dcb0e73217c0331a8, 26d389730aad62ba197498ba43242c4449bba230, 3d178347b40c64ff4322e17bdbaaec4ca3c84e20

**Comment by JordanLeich** [comment]
> Small report about Storm talent at line 61 and 62; the first level value should be 7 if I'm not wrong

You are correct, nice find. Must've been an oversight or typo from the devs. https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/commit/4b023e0c941d02d8dcd40099d1f9f839df8dcf6d

**Comment by ak2yny** [comment]
I re-visited this issue and would like to add a couple of things:
- Level values: They are interpolated for each level (point spent). An even distribution only needs levels 1 and max. (15?) to be set. The reason you might want to add more levels in between in talents files, is that you might want to make damage increase faster on lower levels and slower on higher levels. Each value might need this differently. It's entirely possible to have the level setup completely different for each type. Just the min. and max. must be correct, the rest doesn't matter. (In Storm's case, the difference was minimal, and it was an odd inconsistency, so this felt right to change.)
- Cyclops beam damage: I agree that beam damage does sound odd, even to players that know nothing about the damage type. It would make more sense if it was optic damage, for example. Or just damage, like BloodyMares suggested. I think all DLC characters can have more changes, as we definitely find more problems in their code.
